src/components/__internal__/abstract-custom-field.test.jsx

```diff
@@ -0,0 +1,215 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import AbstractCustomField from './abstract-custom-field.jsx';
+import Icon from '../icon/icon.jsx';
+import calendarSvg from '../../svgs/icon-calendar-fill.svg';
+
+describe('AbstractCustomField', () => {
+  const sharedProps = {
+    label: 'Test label',
+    id: 'test-id',
+  };
+
+  describe('aria props', () => {
+    const ariaProps = { autocomplete: 'list', haspopup: 'listbox' };
+
+    it('sets autocomplete to what is provided', () => {
+      render(<AbstractCustomField {...sharedProps} ariaProps={ariaProps} />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('aria-autocomplete', 'list');
+    });
+
+    it('sets haspopup to what is provided', () => {
+      render(<AbstractCustomField {...sharedProps} ariaProps={ariaProps} />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('aria-haspopup', 'listbox');
+    });
+
+    it('sets the aria-controls to the form control label ID', () => {
+      render(<AbstractCustomField {...sharedProps} ariaProps={ariaProps} />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('aria-controls', 'test-id-label');
+    });
+  });
+
+  describe('className API', () => {
+    it('prioritizes className prop', () => {
+      const { container } = render(<AbstractCustomField {...sharedProps} className="prioritize-me" />);
+      expect(container.firstChild).toHaveClass('prioritize-me');
+    });
+  });
+
```

Comment:
I think we're missing some spec coverage. At least `defaultValue` is missing, maybe others? Can you double-check and fill-in?

Reply:
Ah, we have it on CustomFieldInputText. I didn't move it over. Good catch!